alicloud/data_source_alicloud_db_instances.go

```diff
@@ -0,0 +1,261 @@
+	"regexp"
+)
+
+func dataSourceAlicloudDBInstances() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudDBInstancesRead,
+
+		Schema: map[string]*schema.Schema{
+			"name_regex": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateNameRegex,
+			},
+			"engine": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validateAllowedStringValue([]string{
+					string(MySQL),
+					string(SQLServer),
+					string(PPAS),
+					string(PostgreSQL),
+				}),
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				// please follow the link below to see more details on available statusesplease follow the link below to see more details on available statuses
+				// https://help.aliyun.com/document_detail/26315.html
+			},
+			"instance_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validateAllowedStringValue([]string{
+					"Primary",
+					"Readonly",
+					"Guard",
+					"Temp",
+				}),
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"vswitch_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"connection_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validateAllowedStringValue([]string{
+					"Standard",
+					"Safe",
+				}),
+			},
+			"tags": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateJsonString,
+			},
+			"output_file": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			// Computed values
+			"instances": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"charge_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"instance_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"region_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"create_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"expire_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"engine": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"engine_version": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"net_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"connection_mode": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"instance_class": {
+							Type:     schema.TypeString,
+							Computed: true,
```

Comment:
Please use instance_type to flag its specification and it can be applied to resource 'db_instance' and same as parameter 'instance_type'.

Reply:
parameter name has been changed